examples/reference/custom_components/ReactComponent.md

```diff
@@ -0,0 +1,424 @@
+- **`_import_map`** (dict): This dictionary defines an [import map](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap), allowing you to customize how module specifiers are resolved.
+- **`_stylesheets`** (List[str | PurePath] | None): This optional attribute accepts a list of CSS strings or paths to CSS files. It supports automatic reloading in development environments.
+
+#### `render` Function
+
+The `_esm` attribute must export the `render` function. It accepts the following parameters:
+
+- **`state`**: Manages non-Viewable Parameter state similar to React's [`useState`](https://www.w3schools.com/react/react_usestate.asp) hook. The `render` function is rerun if a Parameter changes.
+- **`data`**: Represents the Viewable Parameters of the component and provides methods to `.watch` for changes and `.send_event` back to Python.
+- **`children`**: Represents the Viewable Parameters of the component. The `render` function is rerun if a child changes.
+- **`model`**: The Bokeh model.
+- **`view`**: The Bokeh view.
+- **`el`**: The HTML element that the component will be rendered into.
+
+Any HTML element returned from the `render` function will be appended to the HTML element (`el`) of the component.
+
+#### Other Lifecycle Methods
+
+DUMMY CONTENT. PLEASE HELP ME DESCRIBE THIS.
+
+- `initialize`: Runs once per widget instance at model initialization, facilitating setup for event handlers or state.
+- `teardown`: Cleans up resources or processes when a widget instance is being removed.
+
+## Usage
+
+### Styling with CSS
+
+Include CSS within the `_stylesheets` attribute to style the component. The CSS is injected directly into the component's HTML.
+
+```python
+import param
+
+import panel as pn
+
+pn.extension()
+
+class CounterButton(pn.ReactComponent):
+
+    value = param.Integer()
+
+    _stylesheets = [
+        """
+        button {
+            background: #0072B5;
+            color: white;
+            border: none;
+            padding: 10px;
+            border-radius: 4px;
+        }
+        button:hover {
+            background: #4099da;
+        }
+        """
+    ]
+
+    _esm = """
+    function App(props) {
+        const [value, setValue] = props.state.value;
+        return (
+            <button onClick={e => setValue(value+1)}>
+            count is {value}
+            </button>
+        );
+    }
+
+    export function render({state}) {
+        return <App state={state}/>;
+    }
+    """
+
+CounterButton().servable()
+```
+
+## Send Events from JavaScript to Python
+
+Events from JavaScript can be sent to Python using the `data.send_event` method. Define a handler in Python to manage these events.
+
+```python
+import param
```

Comment:
Sending events (example below) does not work for me with `ReactComponent` and `PreactComponent`.

Reply:
Your click handler wasn't indented right.